ui/servsettings.py

```diff
@@ -360,3 +386,322 @@ async def get_content(self):
             )
 
         return {"embed": embed}
+
+
+class _RandcharSettingsUI(ServerSettingsMenuBase):
+    # ==== ui ====
+    @disnake.ui.button(label="Set Dice", style=disnake.ButtonStyle.primary)
+    async def select_dice(self, button: disnake.ui.Button, interaction: disnake.Interaction):
+        button.disabled = True
+        await self.refresh_content(interaction)
+        await interaction.send(
+            "Choose a new dice string to roll by sending a message in this channel.",
+            ephemeral=True,
+        )
+        try:
+            input_msg: disnake.Message = await self.bot.wait_for(
+                "message",
+                timeout=60,
+                check=lambda msg: msg.author == interaction.author and msg.channel.id == interaction.channel_id,
+            )
+            self.settings.randchar_dice = input_msg.content
+            with suppress(disnake.HTTPException):
+                await input_msg.delete()
+        except (ValueError, asyncio.TimeoutError):
+            await interaction.send("No valid dice found. Press `Set Dice` to try again.", ephemeral=True)
+        else:
+            await self.commit_settings()
+            await interaction.send("Your dice have been updated.", ephemeral=True)
+        finally:
+            button.disabled = False
+            await self.refresh_content(interaction)
+
+    @disnake.ui.button(label="Set Number of Sets", style=disnake.ButtonStyle.primary)
+    async def select_sets(self, button: disnake.ui.Button, interaction: disnake.Interaction):
+        button.disabled = True
+        await self.refresh_content(interaction)
+        await interaction.send(
+            "Choose a new number of sets to roll by sending a message in this channel.",
+            ephemeral=True,
+        )
+        try:
+            input_msg: disnake.Message = await self.bot.wait_for(
+                "message",
+                timeout=60,
+                check=lambda msg: msg.author == interaction.author and msg.channel.id == interaction.channel_id,
+            )
+            if not 1 <= int(input_msg.content) <= 25:
+                raise ValueError
+            self.settings.randchar_sets = int(input_msg.content)
+            with suppress(disnake.HTTPException):
+                await input_msg.delete()
+        except (ValueError, asyncio.TimeoutError):
+            await interaction.send(
+                "No valid number of sets found. Press `Set Number of Sets` to try again.", ephemeral=True
+            )
+        else:
+            await self.commit_settings()
+            await interaction.send("Your number of sets have been updated.", ephemeral=True)
+        finally:
+            button.disabled = False
+            await self.refresh_content(interaction)
```

Comment:
It might be worth pulling the common "get value form message" logic out into a method (I'd be happy to workshop this on Discord)

Reply:
Will be doing this separately from this PR, after discussion in Discord